commit 3fd1a2ae907bff2d7593c0bb9944aa05eca7b58d
Author: ralcini <roberto.alcini@gmail.com>
Date:   Mon Aug 14 22:43:38 2017 +0200

    fix for issue #624 - force build without openssl (#631)
    
    * fix: now --without-openssl configure flags works
    enh: if openssl support is required and no valid installation is found, now it raise an error
    
    * enh: added warning if building without openssl (iperf_auth disabled)

diff --git a/configure.ac b/configure.ac
index 3c56cb3..79f3869 100644
--- a/configure.ac
+++ b/configure.ac
@@ -101,13 +101,18 @@ AC_CHECK_HEADERS([netinet/sctp.h],
 #endif
 ])
 
-# Check for OPENSSL support
-AX_CHECK_OPENSSL(
-	AC_DEFINE([HAVE_SSL], [1], [OpenSSL Is Available])
-)
-LDFLAGS="$LDFLAGS $OPENSSL_LDFLAGS"
-LIBS="$OPENSSL_LIBS $LIBS"
-CPPFLAGS="$OPENSSL_INCLUDES $CPPFLAGS"
+if test "x$with_openssl" = "xno"; then
+    AC_MSG_WARN( [Building without OpenSSL; disabling iperf_auth functionality.] )
+else
+    # Check for OPENSSL support
+    AX_CHECK_OPENSSL(
+        [ AC_DEFINE([HAVE_SSL], [1], [OpenSSL Is Available]) ],
+	[ AC_MSG_FAILURE([--with-openssl was given, but test for openssl failed]) ]
+    )
+    LDFLAGS="$LDFLAGS $OPENSSL_LDFLAGS"
+    LIBS="$OPENSSL_LIBS $LIBS"
+    CPPFLAGS="$OPENSSL_INCLUDES $CPPFLAGS"
+fi
 
 # Check for TCP_CONGESTION sockopt (believed to be Linux and FreeBSD only)
 AC_CACHE_CHECK([TCP_CONGESTION socket option],
